src/picai_prep/dcm2mha.py

```diff
@@ -26,372 +29,389 @@
-                        remove_c += 1
-            for d in dups[1:]:
-                item = items[d[0]]
-                item['targets'][target] = 'Not selected from duplicate pool due to final randomizing tiebreaker'
-                self.item_log(item, 'tiebreaker (random)')
-                remove_c += 1
-
-        # if target is still None, it is selected. otherwise, the reason for deselection is stated
-        self.info(f"Selected {plural(target_count, 'target')}, removed {remove_c} by tiebreaker.", self.get_history_report())
-
-    def _convert_function(self, args) -> bool:
-        item, target, destination = args
-        # the series is already verified in an earlier step
-        try:
-            image = read_image_series(item['source'])
-        except Exception as e:
-            item['targets'][target] = f'Reading DICOM sequence failed, maybe corrupt data? Error: {e}'
-            self.item_log(item, 'corrupt data')
-            return False
 
-        if self.scan_postprocess_func is not None:
-            image = self.scan_postprocess_func(image)
+class Dicom2MHACase(Case):
+    settings: Dicom2MHASettings = Dicom2MHASettings({}, {})
 
-        try:
-            atomic_image_write(image=image, path=destination, mkdir=True)
-        except Exception as e:
-            item['targets'][target] = f'Error during writing: {e}'
-            self.item_log(item, 'write error')
-            return False
-        return True
+    def __init__(self, input_dir: Path, patient_id: str, study_id: str, paths: List[PathLike]):
+        self.patient_id = patient_id
+        self.study_id = study_id
+        self.series: List[Series] = []
+        self.log = [f'Importing {plural(len(paths), "serie")}']
 
-    def _convert(self):
-        convert_count = 0
-        for item in self.valid_items:
-            for _, status in item['targets'].items():
-                convert_count += 1 if status is None else 0
-
-        self.info(f"Converting {convert_count} DICOM series to MHA.")
-
-        # convert all items with valid targets to mha
-        success_count, skip_count = 0, 0
-        targets = []
-        for item in self.valid_items:
-            for target, status in item['targets'].items():
-                if status is None:  # therefore selected
-                    destination = os.path.join(self.output_dir, item['patient_id'], target + '.mha')
-                    if os.path.exists(destination):
-                        item['targets'][target] = 'Skipped, target output path already exists'
-                        self.item_log(item, 'skipped')
-                        skip_count += 1
-                    else:
-                        targets.append((item, target, destination))
+        fullpaths = set()
+        for path in paths:
+            full_path = input_dir / path
+            serie = Series(full_path, patient_id, study_id)
+            try:
+                if not full_path.exists():
```

Comment:
I think so, because its warning for an underlying problem (archives are always generated, if there are duplicates during such generation, that's very fishy)